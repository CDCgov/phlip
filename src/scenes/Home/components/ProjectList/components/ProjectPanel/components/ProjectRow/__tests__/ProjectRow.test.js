import React from 'react'
import { ProjectRow } from '../index'
import { shallow } from 'enzyme'

const props = {
  project: {
    createdBy: 'Admin',
    createdByEmail: 'test@test.test',
    createdById: 1,
    dateCreated: '2019-04-08T11:43:46.3034169',
    dateLastEdited: '2019-04-18T13:27:51.454041',
    id: 2,
    lastEditedBy: 'Aleksandra Zaryanova',
    lastUsersCheck: 1555606985567,
    name: 'PROJECT',
    projectUsers: [],
    projectJurisdictions: [],
    type: 1
  },
  isCoder: false,
  bookmarked: false,
  toggleBookmark: jest.fn()
}

const setup = (other = {}) => {
  return shallow(
    <ProjectRow {...props} {...other} />
  )
}

describe('Home - ProjectList - ProjectPanel - ProjectRow component', () => {
  test('should render correctly', () => {
    expect(shallow(<ProjectRow {...props} />)).toMatchSnapshot()
  })
  
  describe('bookmark icon', () => {
    test('should be `bookmark` if the project is bookmarked', () => {
      const wrapper = setup({ bookmarked: true }).find('IconButton').at(0).childAt(0)
      expect(wrapper.text()).toContain('bookmark')
    })
    
    test('should be `bookmark_border` if the project is not bookmarked', () => {
      const wrapper = setup().find('IconButton').at(0).childAt(0)
      expect(wrapper.text()).toContain('bookmark_border')
    })
    
    test('should be grey if the project is not bookmarked', () => {
      const wrapper = setup().find('IconButton').at(0)
      expect(wrapper.prop('color')).toEqual('#757575')
    })
    
    test('should be orange if the project is bookmarked', () => {
      const wrapper = setup({ bookmarked: true }).find('IconButton').at('0')
      expect(wrapper.prop('color')).toEqual('#fdc43b')
    })
  
    test('should call props.toggleBookmark when the bookmark icon is clicked', () => {
      const spy = jest.spyOn(props, 'toggleBookmark')
      let wrapper = setup()
      wrapper.find('IconButton').at(0).simulate('click', { stopPropagation: () => undefined })
      wrapper.update()
      expect(spy).toHaveBeenCalled()
    })
  })
})
