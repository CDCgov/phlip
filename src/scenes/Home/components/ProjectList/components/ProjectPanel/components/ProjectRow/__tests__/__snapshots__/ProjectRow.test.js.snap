// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home - ProjectList - ProjectPanel - ProjectRow component should render correctly 1`] = `
<FlexGrid
  align="stretch"
  circular={false}
  container={true}
  flex={true}
  justify="stretch"
  padding={0}
  raised={false}
  style={
    Object {
      "minHeight": 60,
    }
  }
  type="row"
>
  <FlexGrid
    align="center"
    circular={false}
    container={true}
    flex={false}
    justify="stretch"
    padding="0 0 0 24px"
    raised={false}
    style={
      Object {
        "width": "24px",
      }
    }
    type="row"
  >
    <IconButton
      aria-label="Bookmark this project"
      color="#757575"
      disabled={false}
      iconSize={24}
      id="bookmark-project-2"
      onClick={[Function]}
      tooltipText="Bookmark project"
    >
      bookmark_border
    </IconButton>
  </FlexGrid>
  <FlexGrid
    align="center"
    circular={false}
    container={true}
    flex={false}
    justify="stretch"
    padding="0 12px"
    raised={false}
    style={
      Object {
        "flexBasis": "32%",
      }
    }
    type="row"
  >
    <span>
      PROJECT
    </span>
  </FlexGrid>
  <FlexGrid
    align="center"
    circular={false}
    container={true}
    flex={true}
    justify="stretch"
    padding={0}
    raised={false}
    style={
      Object {
        "flexBasis": "30%",
      }
    }
    type="row"
  >
    <FlexGrid
      align="stretch"
      circular={false}
      container={false}
      flex={false}
      justify="stretch"
      padding={0}
      raised={false}
      style={
        Object {
          "width": "49%",
        }
      }
      type="column"
    >
      <span
        style={
          Object {
            "color": "black",
          }
        }
      >
        Date Last Edited:
         
      </span>
      <span>
        4/18/2019
      </span>
    </FlexGrid>
    <span
      style={
        Object {
          "width": "2%",
        }
      }
    />
    <FlexGrid
      align="stretch"
      circular={false}
      container={false}
      flex={false}
      justify="stretch"
      padding={0}
      raised={false}
      style={
        Object {
          "width": "49%",
        }
      }
      type="column"
    >
      <span
        style={
          Object {
            "color": "black",
          }
        }
      >
        Last Edited By:
         
      </span>
      Aleksandra Zaryanova
    </FlexGrid>
  </FlexGrid>
  <FlexGrid
    align="center"
    circular={false}
    container={true}
    flex={true}
    justify="flex-end"
    padding="0 44px 0 0"
    raised={false}
    style={
      Object {
        "flexBasis": "30%",
      }
    }
    type="row"
  >
    <WithTheme(Button)
      aria-label="Code project"
      component={[Function]}
      listButton={true}
      raised={false}
      style={
        Object {
          "borderRadius": 3,
        }
      }
      to={
        Object {
          "pathname": "/project/2/code",
        }
      }
      value="Code"
    />
    <span
      style={
        Object {
          "width": "24px",
        }
      }
    />
    <WithTheme(Button)
      aria-label="Validate project"
      component={[Function]}
      listButton={true}
      raised={false}
      style={
        Object {
          "borderRadius": 3,
        }
      }
      to={
        Object {
          "pathname": "/project/2/validate",
        }
      }
      value="Validate"
    />
  </FlexGrid>
</FlexGrid>
`;
